src/ServiceControlInstaller.Engine/Instances/MonitoringInstance.cs

```diff
@@ -290,5 +292,17 @@ public void SetupInstance()
                 ReportCard.Errors.Add(ex.Message);
             }
         }
+
+        public void UpgradeTransportSeam()
+        {
+            //Legacy ASB Forwarding Topology should be migrated to the ASBS transport seam
+            if (TransportPackage.Name == TransportNames.AzureServiceBusForwardingTopologyDeprecated)
```

Comment:
Would it make sense to put an `AutoMigrateTo` property on `TransportInfo`? That way this code could check for a value and migrate without knowing specific types, and if we want to do this again, it's just metadata.

Reply:
How about now?